client/order/epochqueue.go

```diff
@@ -37,13 +39,23 @@ func NewEpochQueue() *EpochQueue {
 
 // Reset clears the epoch queue. This should be called when a new epoch begins.
 func (eq *EpochQueue) Reset() {
+	atomic.StoreUint64(&eq.epoch, 0)
+
 	eq.mtx.Lock()
 	eq.orders = make(map[order.OrderID]*epochOrder)
 	eq.mtx.Unlock()
 }
 
 // Enqueue appends the provided order note to the epoch queue.
 func (eq *EpochQueue) Enqueue(note *msgjson.EpochOrderNote) error {
+	epoch := atomic.LoadUint64(&eq.epoch)
+
+	// Ensure the order received is of the current epoch.
+	if epoch != 0 && note.Epoch != epoch {
+		return fmt.Errorf("epoch mismatch: expected %d, got %d",
+			epoch, note.Epoch)
+	}
+
```

Comment:
Looks like `eq.mtx` is not locked on the access `_, ok := eq.orders[oid]` below at https://github.com/decred/dcrdex/pull/188/files#diff-6555aa5b6944be6d9b47df699e4fa894R68